packages/devtools/lib/src/trees.dart

```diff
@@ -42,15 +42,29 @@ class TreeNode<T extends TreeNode<T>> {
     if (_root != null) {
       return _root;
     }
+
+    // Store nodes we have visited so we can cache the root value for each one
+    // once we find the root.
+    // ignore: prefer_collection_literals
+    final Set<T> visited = Set.from([this]);
```

Comment:
[nit] `Set.of` might be a little more appropriate here. `Set.from` is intended for cases where you need to cast elements.